test/test_line_profiler.py

```diff
@@ -1,32 +1,40 @@
+import re
+
+import numpy as np
 import pytest
 import torch
-import numpy as np
+from pytorch_memlab import (LineProfiler, clear_global_line_profiler, profile,
+                            profile_every, set_target_gpu)
 
-from pytorch_memlab import LineProfiler, profile, profile_every, set_target_gpu, clear_global_line_profiler
 
 def test_display():
 
-    def work():
-        # comment
+    def main():
         linear = torch.nn.Linear(100, 100).cuda()
-        linear_2 = torch.nn.Linear(100, 100).cuda()
-        linear_3 = torch.nn.Linear(100, 100).cuda()
+        part1()
+        part2()
 
-    def work_3():
+    def part1():
         lstm = torch.nn.LSTM(1000, 1000).cuda()
+        subpart11()
 
-    def work_2():
-        # comment
+    def part2():
+        linear_2 = torch.nn.Linear(100, 100).cuda()
+        linear_3 = torch.nn.Linear(100, 100).cuda()
+
+    def subpart11():
         linear = torch.nn.Linear(100, 100).cuda()
         linear_2 = torch.nn.Linear(100, 100).cuda()
         linear_3 = torch.nn.Linear(100, 100).cuda()
-        work_3()
-    
-    with LineProfiler(work, work_2) as prof:
-        work()
-        work_2()
 
-    return prof.display()
+    with LineProfiler(subpart11, part2) as prof:
+        main()
+
+    s = str(prof.display())  # cast from line_records.RecordsDisplay
+    assert re.search("## .*subpart11", s)
+    assert "def subpart11():" in s
+    assert re.search("## .*part2", s)
+    assert "def part2():" in s
```

Comment:
and we don't want functions like `main` and `part1` to appear here.
`assert 'main' not in s` should suffice?